### ISSUE TITLE
[BUG] Subsume instance creation failure causes underlying target resource to be deleted

### ISSUE THREAD

**Issue by claassen** [cloudfoundry-incubator/csb-brokerpak-azure#10]
It seems that when the creation of a subsume instance fails, it causes the broker to trigger a deprovision which actually deletes the underlying DB that was being subsumed. I assume it does this is a way to rollback partially created resources when something fails which makes sense for new resources, but not when subsuming.

I saw this happen when the subsume instance failed with the following error:

```
Service [rates-sql-uat] creation/update failed: Error: Error waiting on create/update future for SQL Failover Group "rates-sql-uat-failover" (Resource Group "MFC-CAC-PCF-PSB", Server "azugessqlprod"): Future#WaitForCompletion: context has been cancelled: StatusCode=200 -- Original Error: context deadline exceeded on main.tf line 56, in resource "azurerm_sql_failover_group" "failover_group": 56: resource "azurerm_sql_failover_group" "failover_group" { exit status 1
```

I think a solution would be for subsume instance creation to not trigger a deprovision when it fails.

**Comment by erniebilling** [comment]
Ugh. This is automatic behavior by terraform when a step in the process fails and I don't immediately see a way to prevent it at the moment. Investigating.

**Comment by erniebilling** [comment]
Little more investigation here - this behavior is limited to fail over group subsume (since it imports multiple resources.) Terraform believed it had successfully created the primary and secondary db and then couldn't create the failover group and rolled back. Still looking for way to tell terraform not to do rollback.

**Comment by erniebilling** [comment]
Root cause here is that applying any change to a secondary db causes the secondary DB to be re-created. Issue opened and active in terraform azurerm provider - https://github.com/terraform-providers/terraform-provider-azurerm/issues/11024

**Comment by claassen** [comment]
Just to clarify the issue here, the fact that Terraform recreates the secondary DB causes it to delete the other resources as well when rolling back due to a failure?

**Comment by pivotal-marcela-campo** [comment]
Tested the terraform provider fix and unfortunately it's still failing. The provider is expecting an extra parameter to be passed in on updates to the secondary, otherwise it fails. When passing in the parameter, it goes back to recreating the database.

Raised a new issue with them [here](https://github.com/terraform-providers/terraform-provider-azurerm/issues/11681).

**Comment by aristosvo** [comment]
This is fixed now in the latest release of the Terraform Azure provider, `2.59.0`

**Comment by pivotal-marcela-campo** [comment]
Thanks @aristosvo, this worked fine.
Many thanks for all your help!

**Comment by pivotal-marcela-campo** [comment]
Work to consume the new provider version and other changes needed on our side has been completed. This fix will be included in our next release.

**Comment by tsuiala** [comment]
Hi @erniebilling,
Shall we reopen this case as we found the latest release doesn't resolve [issue 53](https://github.com/cloudfoundry-incubator/csb-brokerpak-azure/issues/53) please?
Thanks a lots.
